spec/rubocop/cop/performance/caller_locations_spec.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe RuboCop::Cop::Performance::CallerLocations do
+  subject(:cop) { described_class.new }
+
+  it "doesn't register an offense when caller is called" do
+    inspect_source(cop, 'caller')
+    expect(cop.messages).to be_empty
+  end
+
+  it 'registers an offense when :[] is called on caller' do
+    inspect_source(cop, 'caller[0]')
+    expect(cop.messages).to eq([described_class::MSG])
```

Comment:
We would normally assert:

```ruby
expect(cop.offenses.size).to eq(1)
```

Asserting messages is generally only done for cops that have messages that change depending on context. 🙂